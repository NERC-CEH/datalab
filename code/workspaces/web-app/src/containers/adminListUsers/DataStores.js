import React from 'react';
import { TYPE_NAME, TYPE_NAME_PLURAL } from '../dataStorage/DataStorageContainer';
import ResourceStackCards from './ResourceStackCards';

function DataStores({ dataStores }) {
  return (
    <ResourceStackCards resources={dataStores} typeName={TYPE_NAME} typeNamePlural={TYPE_NAME_PLURAL} />
  );
}

export default DataStores;
