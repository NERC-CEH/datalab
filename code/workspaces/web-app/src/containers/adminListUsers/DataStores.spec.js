import React from 'react';
import { render } from '@testing-library/react';
import DataStores from './DataStores';

jest.mock('./ResourceStackCards', () => props => (<>ResourceStackCards Mock {JSON.stringify(props)}</>));

describe('DataStores', () => {
  const dataStores = [{
    key: 'test-key',
    name: 'Test Name',
  }];
  const props = {
    dataStores,
  };

  it('renders to match snapshot passing correct props to children', () => {
    expect(render(<DataStores {...props} />).container).toMatchSnapshot();
  });
});
