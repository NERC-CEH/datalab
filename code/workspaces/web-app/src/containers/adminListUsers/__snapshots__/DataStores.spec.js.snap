// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataStores renders to match snapshot passing correct props to children 1`] = `
<div>
  ResourceStackCards Mock 
  {"resources":[{"key":"test-key","name":"Test Name"}],"typeName":"Data Store","typeNamePlural":"Data Stores"}
</div>
`;
